verification_service/main.py

```diff
@@ -0,0 +1,309 @@
+import logging
+import os
+import re
+import requests
+import yaml
+from time import sleep
+from prometheus_client import multiprocess, CollectorRegistry, Counter
+from flask import Flask, request, jsonify
+from threading import Thread
+
+from common.event_processor import EventProcessor
+
+from common.constants import (
+    project_events,
+    user_events,
+    issue_events,
+    issue_note_events,
+    group_events,
+    event_types,
+)
+
+LOGLEVEL = os.environ.get('LOGLEVEL', 'WARNING').upper()
+logging.basicConfig(
+    level=LOGLEVEL, format="%(asctime)s - %(levelname)s - Verification service: %(message)s"
+)
+
+# Prometheus metrics
+prometheus_multiproc_dir = "prometheus_multiproc_dir"
+
+os.makedirs(prometheus_multiproc_dir, exist_ok=True)
+
+registry = CollectorRegistry()
+multiprocess.MultiProcessCollector(registry)
+
+processed_events_total = Counter(
+    "verification_service_processed_events_total", "Total number of processed events"
+)
+verified_events_total = Counter(
+    "verification_service_verified_events_total", "Total number of verified events"
+)
+
+verification_failures_total = Counter(
+    "verification_service_verification_failures_total",
+    "Total number of verification failures",
+)
+gitlab_api_calls_total = Counter(
+    "verification_service_gitlab_api_calls_total",
+    "Total number of GitLab API calls",
+    ["status"],
+)
+snippet_check_events_total = Counter(
+    "verification_service_snippet_check_events_total",
+    "Total number of snippet check events processed",
+)
+
+# Flask app for verification at a later point in the Spampibian pipeline
+# This is used to verify individual snippets that first need to be
+# retrieved from GitLab by the retrieval service.
+app = Flask(__name__)
+
+CONTENT_TYPE_LATEST = str("text/plain; version=0.0.4; charset=utf-8")
+
+
+# User email domain verification endpoint
+@app.route("/verify_email", methods=["POST"])
+def verify_email():
```

Comment:
**question:** this doesn't seem to be called anywhere?

Reply:
Oversight on my end - that should be called by the retrieval service for group and snippet events. I will add that bit in.